io/pci/pci_info_lscfg.py

```diff
@@ -0,0 +1,121 @@
+
+                # Slot Match
+                if 'YL' in cfg_output:
+                    sys_slot = pci.get_slot_from_sysfs(pci_addr)
+                    if sys_slot:
+                        sys_slot = sys_slot.strip('\0')
+                    cfg_slot = cfg_output['YL']
+                    self.log.info("Slot from sysfs: %s", sys_slot)
+                    self.log.info("Slot from lscfg: %s", cfg_slot)
+                    if sys_slot in [sys_slot, cfg_slot[:cfg_slot.rfind('-')]]:
+                        self.log.info("=======>>> slot matches perfectly\n\n")
+                    else:
+                        error.append(pci_addr + "-> slot")
+                        self.log.info("--->>Slot Numbers not Matched\n\n")
+                else:
+                    self.log.error("Slot info not available in cfg output\n")
+
+                # Device ID match
+                sys_pci_id_output = pci.get_pci_id_from_sysfs(pci_addr)
+                cfg_dev_id = cfg_output['pci_id'][4:]
+                sysfs_dev_id = sys_pci_id_output[5:-10]
+                sysfs_sdev_id = sys_pci_id_output[15:]
+                self.log.info("Device ID from sysfs: %s", sysfs_dev_id)
+                self.log.info("Sub Device ID from sysfs: %s", sysfs_sdev_id)
+                self.log.info("Device ID from cfg: %s", cfg_dev_id)
+                if cfg_dev_id == sysfs_sdev_id or cfg_dev_id == sysfs_dev_id:
+                    self.log.info("=======>>Device ID Match Success\n\n")
+                else:
+                    self.log.error("----->>Device ID did not Match\n\n")
+                    error.append(pci_addr + "-> Device_id")
+
+                # Subvendor ID Match
+                sysfs_subvendor_id = sys_pci_id_output[10:-5]
+                cfg_subvendor_id = cfg_output['pci_id'][:4]
+                self.log.info("Subvendor ID frm sysfs: %s", sysfs_subvendor_id)
+                self.log.info("Subvendor ID from cfg : %s", cfg_subvendor_id)
+                if sysfs_subvendor_id == cfg_subvendor_id:
+                    self.log.info("======>>>Subvendor ID Match Success\n\n")
+                else:
+                    self.log.error("---->>Subvendor_id Not Matched\n\n")
+                    error.append(pci_addr + "-> Subvendor_id")
+
+                # PCI ID Match
+                lspci_pci_id = pci.get_pci_id(pci_addr)
+                self.log.info(" PCI ID from Sysfs: %s", sys_pci_id_output)
+                self.log.info("PCI ID from Cfg : %s", lspci_pci_id)
+
+                if sys_pci_id_output == lspci_pci_id:
+                    self.log.info("======>>>> All PCI ID match Success\n\n")
+                else:
+                    self.log.error("---->>>PCI info Did not Matches\n\n")
+                    error.append(pci_addr + "-> pci_id")
+
+                # PCI Config Space Check
+                if process.system("lspci -xxxx -s %s" % pci_addr,
+                                  ignore_status=True, sudo=True):
```

Comment:
can it be more moduler .. meaning each match will have separate function defined ?
def test_pci_match(..
def test_vendorid_match(..
def test_slot_match(

or may be I see this part of the code is redundant .. can this be optimised
                self.log.info("Subvendor ID frm sysfs: %s", sysfs_subvendor_id)
                self.log.info("Subvendor ID from cfg : %s", cfg_subvendor_id)
                if sysfs_subvendor_id == cfg_subvendor_id:
                    self.log.info("======>>>Subvendor ID Match Success\n\n")
                else:
                    self.log.error("---->>Subvendor_id Not Matched\n\n")
                    error.append(pci_addr + "-> Subvendor_id")